types/address.go

```diff
@@ -256,6 +256,21 @@ func (aa AccAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// LengthPrefixAddress prefixes the address bytes with its length, this is used
+// for variable-length components in store keys. Note: All addresses should be
+// max 255 bytes, or else this function panics.
+func LengthPrefixAddress(bz []byte) []byte {
+	if len(bz) == 0 {
+		return bz
+	}
+
+	if len(bz) > 255 {
+		panic("address length should be max 255 bytes")
+	}
+
+	return append([]byte{byte(len(bz))}, bz...)
```

Comment:
Is this the correct way? Should we not instead take the length and encode it into a fixed size integer? For example, if I wanted to decode this, I have no idea how many bytes to read to get the length.

Reply:
If we assume that bz is max 255 bytes long, then we're exactly doing what you're saying: encode it into a fixed-size (1-byte) integer.

To decode, we read the first byte, it's `N`, and then read the next `N` bytes. Code [here](https://github.com/cosmos/cosmos-sdk/pull/8363/files#diff-c06d258288f3f4e86f19a61cd7a01fce7b5203b5c8aeb7019461bacf7944bb26R195-R197).

Reply:
Looks good then.